components/service/glean/src/main/java/mozilla/components/service/glean/storages/TimespansStorageEngine.kt

```diff
@@ -0,0 +1,142 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean.storages
+
+import android.annotation.SuppressLint
+import android.os.SystemClock
+import mozilla.components.service.glean.CommonMetricData
+
+import mozilla.components.support.base.log.logger.Logger
+import java.util.concurrent.TimeUnit
+
+/**
+ * Enumeration of different resolutions supported by
+ * the Timespan metric type.
+ */
+enum class TimespanResolution {
```

Comment:
nit: this should be renamed to `TimespanTimeUnit` to better match the naming used in the spec.